feat: added accessor protocol support to `stats/base/meanpn` (#5908)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/meanpn/README.md

var meanpn = require( '@stdlib/stats/base/meanpn' );
```

#### meanpn( N, x, strideX )

> Computes the [arithmetic mean][arithmetic-mean] of a strided array `x` using a two-pass error correction algorithm.

Reviewer: ```suggestion
Computes the [arithmetic mean][arithmetic-mean] of a strided array using a two-pass error correction algorithm.
```

-   **N**: number of indexed elements.
-   **x**: input [`Array`][mdn-array] or [`typed array`][mdn-typed-array].
-   **strideX**: stride length for `x`.

> The `N` and `strideX` parameters determine which elements in `x` are accessed at runtime. For example, to compute the [arithmetic mean][arithmetic-mean] of every other element in `x`,

Reviewer: ```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to compute the [arithmetic mean][arithmetic-mean] of every other element in the input array,
```

```javascript

> var floor = require( '@stdlib/math/base/special/floor' );

Reviewer: @dhruvilmehta Would you also mind updating the examples throughout this package to no longer use `floor`?

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

var v = meanpn( 4, x1, 2 );
// returns 1.25
```

#### meanpn.ndarray( N, x, strideX, offsetX )

Computes the [arithmetic mean][arithmetic-mean] of a strided array using a two-pass error correction algorithm and alternative indexing semantics.

```javascript
var x = [ 1.0, -2.0, 2.0 ];

var v = meanpn.ndarray( x.length, x, 1, 0 );
// returns ~0.33333
```

The function has the following additional parameters:

-   **offsetX**: starting index for `x`.

> While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to calculate the [arithmetic mean][arithmetic-mean] for every other value in the strided array starting from the second element

Reviewer: ```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to calculate the [arithmetic mean][arithmetic-mean] for every other element in the strided array starting from the second element
```

> While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to calculate the [arithmetic mean][arithmetic-mean] for every other element in the strided array starting from the second element

Reviewer: ```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to calculate the [arithmetic mean][arithmetic-mean] for every other element in the strided array starting from the second element
```

The function has the following additional parameters:

-   **offsetX**: starting index for `x`.

While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to calculate the [arithmetic mean][arithmetic-mean] for every other value in `x` starting from the second value

> 

Reviewer: See other similar packages, but we also tend to add a note concerning accessor protocol support below.

## Notes

-   If `N <= 0`, both functions return `NaN`.
-   Both functions support array-like objects having getter and setter accessors for array element access (e.g., [`@stdlib/array/base/accessor`][@stdlib/array/base/accessor]).
-   Depending on the environment, the typed versions ([`dmeanpn`][@stdlib/stats/base/dmeanpn], [`smeanpn`][@stdlib/stats/base/smeanpn], etc.) are likely to be significantly more performant.

> </section>

Reviewer: The example below still needs updating.

Reviewer: See `examples/index.js`.

> var randu = require( '@stdlib/random/base/randu' );
> var round = require( '@stdlib/math/base/special/round' );
> var Float64Array = require( '@stdlib/array/float64' );

Reviewer: ```suggestion
```
